### PR TITLE
apply enable_public_net_ipv4 and enable_public_net_ipv6 to autoscaler configuration

### LINKED ISSUE #372: disable public ipv4 / ipv6 ignored on autoscaled nodes
Hi, the `enable_public_net_ipv4` parameter seems to be ignored on autoscaled nodes. 
I have used the example config and set `enable_public_net_ipv4` & `enable_public_net_ipv6` to false as well as configured a private network for the static nodes which works flawlessly.  
Autoscaled nodes are also set up correctly to use the private network but they are assigned a public ipv4 & ipv6 address.

Awesome tool btw :)

My config:
```yaml
hetzner_token: --
cluster_name: test
kubeconfig_path: "./kubeconfig"
k3s_version: v1.26.4+k3s1
public_ssh_key_path: "~/.ssh/testcluster.pub"
private_ssh_key_path: "~/.ssh/testcluster"
use_ssh_agent: false
ssh_allowed_networks:
  - 0.0.0.0/0 # ensure your current IP is included in the range
api_allowed_networks:
  - 0.0.0.0/0 # ensure your current IP is included in the range
disable_flannel: false # set to true if you want to install a different CNI
schedule_workloads_on_masters: false
enable_public_net_ipv4: false # default is true
enable_public_net_ipv6: false # default is true
datastore:
  mode: etcd # etcd (default) or external
#  external_datastore_endpoint: postgres://....
masters_pool:
  instance_type: cx22
  instance_count: 1
  location: fsn1
worker_node_pools:
  - name: small-static
    instance_type: cx22
    instance_count: 1
    location: hel1
  - name: small-autoscaled
    instance_type: cx22
    instance_count: 1
    location: fsn1
    autoscaling:
      enabled: true
      min_instances: 0
      max_instances: 1

existing_network: my_private_network
enable_encryption: true
post_create_commands:
  - timedatectl set-timezone Europe/Berlin
  - >
    printf "network## {config## disabled}" |
    sed 's/##/:/g' > /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg
  - >
    printf "network##\n  version## 2\n  renderer## networkd\n  ethernets##\n    enp7s0##\n      dhcp4## true\n      nameservers##\n        addresses## [8.8.8.8, 8.8.4.4]\n      routes##\n        - to## default\n          via## 10.0.0.1" |
    sed 's/##/:/g' > /etc/netplan/50-cloud-init.yaml
  - netplan generate
  - netplan apply
  - apt update
  - apt upgrade -y
  - apt install nfs-common -y
  - apt autoremove -y

```

### PR DESCRIPTION
**Pull request by Funzinator**
fixes #372

### CODE REVIEW

File: src/kubernetes/software/cluster_autoscaler.cr [hunk 1]
```diff
@@ -131,6 +131,8 @@ class Kubernetes::Software::ClusterAutoscaler
     set_container_environment_variable(autoscaler_container, "HCLOUD_FIREWALL", settings.cluster_name)
     set_container_environment_variable(autoscaler_container, "HCLOUD_SSH_KEY", settings.cluster_name)
     set_container_environment_variable(autoscaler_container, "HCLOUD_NETWORK", (settings.existing_network || settings.cluster_name))
+    set_container_environment_variable(autoscaler_container, "HCLOUD_PUBLIC_IPV4", settings.enable_public_net_ipv4 ? "true" : "false")
+    set_container_environment_variable(autoscaler_container, "HCLOUD_PUBLIC_IPV6", settings.enable_public_net_ipv6 ? "true" : "false")
```
Comment (vitobotta) [inline src/kubernetes/software/cluster_autoscaler.cr:135]: Hi, thanks for the PR! Those settings are already booleans, so you should be able to do 

```suggestion
    set_container_environment_variable(autoscaler_container, "HCLOUD_PUBLIC_IPV4", settings.enable_public_net_ipv4.to_s)
    set_container_environment_variable(autoscaler_container, "HCLOUD_PUBLIC_IPV6", settings.enable_public_net_ipv6.to_s)
```

**Review by vitobotta — CHANGES_REQUESTED** [review]
Hi! Thanks a lot. See comment - if you can test that change it would be great :)

**Comment by Funzinator** [comment]
Ah, learned something new about Crystal :-)...
That works equally well. Amended my commit.

**Comment by vitobotta** [comment]
Thanks!

**Comment by vitobotta** [comment]
I also merged it into the v2 branch as well :)

**Comment by benkrej** [comment]
Thank you for fixing!

**Pull request merged**